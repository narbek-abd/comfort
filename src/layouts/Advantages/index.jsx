import React from 'react';
import { SContainer, SectionTitle } from "../../globalStyle";
import delivery  from '../../assets/icons/delivery.svg';
import cashback  from '../../assets/icons/cashback.svg';
import premium  from '../../assets/icons/premium.svg';
import support  from '../../assets/icons/support.svg';

import {
  SAdvantages,
  SInner,
SItem,
} from "./style";
import { Icon } from '../../components/Icon';

const Advantages = () => {
  return (
    <SAdvantages>
    <SectionTitle>What Shopex Offer!</SectionTitle>
      <SContainer>
        <SInner>
          <SItem>
            <img src={delivery} alt="" />

            <h3>24/7 Support</h3>
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Massa purus gravida.</p>
          </SItem>

          <SItem>
            <img src={cashback} alt="" />

            <h3>24/7 Support</h3>
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Massa purus gravida.</p>
          </SItem>

          <SItem>
            <img src={premium} alt="" />

            <h3>24/7 Support</h3>
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Massa purus gravida.</p>
          </SItem>

          <SItem>
            <img src={support} alt="" />

            <h3>24/7 Support</h3>
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Massa purus gravida.</p>
          </SItem>
        </SInner>
      </SContainer>
      
    </SAdvantages>
  );
};

export default Advantages;