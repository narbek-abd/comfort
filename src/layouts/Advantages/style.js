import styled from "styled-components/macro";

export const SAdvantages = styled.section`
margin-top: 58px;
 `;

export const SInner = styled.div`
margin-top: 116px;
	display: flex;
	flex-wrap: wrap;
	justify-content: space-between;
`;

export const SItem = styled.div`
	width: 270px;
	text-align: center;
	h3 {
		margin-top: 27px;
		font-family: Josefin Sans;
		font-weight: 600;
		font-size: 22px;
		line-height: 22px;
		color: var(--color-dark-blue);
	}

	p {
		margin-top: 20px;
		font-family: Lato;
		font-weight: bold;
		font-size: 16px;
		line-height: 26px;
		color: rgba(26, 11, 91, 0.3);
	}

	@media only screen and (max-width:768px) {
		width: 49%;
		&:nth-child(n+3) {
			margin-top: 50px;
		}
	}

	@media only screen and (max-width:576px) {
		width: 100%;
		&:not(:first-child) {
			margin-top: 50px;
		}
	}
`;
