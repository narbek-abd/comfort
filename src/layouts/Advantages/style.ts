import styled from "styled-components";

export const Advantages = styled.section`
	margin-top: 58px;
`;

export const Inner = styled.div`
	margin-top: 116px;
	display: flex;
	flex-wrap: wrap;
	justify-content: space-between;
`;

export const Item = styled.div`
	width: 270px;
	text-align: center;
	h3 {
		margin-top: 27px;
		font-family: Josefin Sans;
		font-weight: 600;
		font-size: 22px;
		line-height: 22px;
		color: var(--color-dark-blue);
	}

	p {
		margin-top: 20px;
		font-family: Lato;
		font-weight: bold;
		font-size: 16px;
		line-height: 26px;
		color: rgba(26, 11, 91, 0.3);
	}

	@media only screen and (max-width: 768px) {
		width: 49%;
		&:nth-child(n + 3) {
			margin-top: 50px;
		}
	}

	@media only screen and (max-width: 576px) {
		width: 100%;
		&:not(:first-child) {
			margin-top: 50px;
		}
	}
`;
