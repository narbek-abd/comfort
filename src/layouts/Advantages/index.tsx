import React from "react";
import * as G from "../../globalStyle";
import delivery from "../../assets/icons/delivery.svg";
import cashback from "../../assets/icons/cashback.svg";
import premium from "../../assets/icons/premium.svg";
import support from "../../assets/icons/support.svg";

import * as S from "./style";
import Icon from "../../components/Icon";

const Advantages = () => {
  return (
    <S.Advantages>
      <G.SectionTitle>What Shopex Offer!</G.SectionTitle>
      
      <G.Container>
        <S.Inner>
          <S.Item>
            <img src={delivery} alt="" />

            <h3>24/7 Support</h3>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Massa
              purus gravida.
            </p>
          </S.Item>

          <S.Item>
            <img src={cashback} alt="" />

            <h3>24/7 Support</h3>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Massa
              purus gravida.
            </p>
          </S.Item>

          <S.Item>
            <img src={premium} alt="" />

            <h3>24/7 Support</h3>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Massa
              purus gravida.
            </p>
          </S.Item>

          <S.Item>
            <img src={support} alt="" />

            <h3>24/7 Support</h3>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Massa
              purus gravida.
            </p>
          </S.Item>
        </S.Inner>
      </G.Container>
    </S.Advantages>
  );
};

export default Advantages;
