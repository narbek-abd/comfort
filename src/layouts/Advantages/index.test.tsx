import { screen, waitFor, render } from "@testing-library/react";
import Advantages from "./index";

describe("Advantages component", () => {

	test("should display component", async () => {
		const {container} = render(<Advantages />);

		expect(container).toMatchSnapshot(); 
	});
});
