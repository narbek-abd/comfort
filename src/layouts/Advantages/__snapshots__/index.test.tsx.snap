// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advantages component should display component 1`] = `
<div>
  <section
    class="sc-iCfMLu bMNXMi"
  >
    <h2
      class="sc-gsDKAQ eeQIaO"
    >
      What Shopex Offer!
    </h2>
    <div
      class="sc-bdvvtL gFjKpS"
    >
      <div
        class="sc-furwcr jmgjLu"
      >
        <div
          class="sc-pVTFL hHPiwM"
        >
          <img
            alt=""
            src="delivery.svg"
          />
          <h3>
            24/7 Support
          </h3>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Massa purus gravida.
          </p>
        </div>
        <div
          class="sc-pVTFL hHPiwM"
        >
          <img
            alt=""
            src="cashback.svg"
          />
          <h3>
            24/7 Support
          </h3>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Massa purus gravida.
          </p>
        </div>
        <div
          class="sc-pVTFL hHPiwM"
        >
          <img
            alt=""
            src="premium.svg"
          />
          <h3>
            24/7 Support
          </h3>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Massa purus gravida.
          </p>
        </div>
        <div
          class="sc-pVTFL hHPiwM"
        >
          <img
            alt=""
            src="support.svg"
          />
          <h3>
            24/7 Support
          </h3>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Massa purus gravida.
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
